### PR TITLE
Run tests on GitHub Actions

### LINKED ISSUE #4719: Proposal: Consider running tests on GitHub Actions in addition to Circle CI
### Describe the problem and steps to reproduce it:

This is not a bug but rather a proposal: use GitHub Actions CI _in addition_ to Circle CI.

### What happened?

1. Sign into Circle CI and grant it access to GitHub account in order to run jobs on PRs
2. Go Circle CI access settings on https://github.com/settings/connections/applications/78a2ba87f071c28e65bb
3. Observe that Circle CI requires access to private repositories on GitHub without any way to revoke this access while keeping 

image

### What did you expect to happen?

Users have a choice to run CI jobs (tests) on a CI service following follows least-privilege principle (e.g., without granting "Full control of private repositories").

### Anything else we should know?

Back in 2020 I helped BCD try out GitHub Actions for their needs (in addition to Travis CI at the time), after a while BCD switched to GitHub Actions exclusively. Here I propose to run tests on GitHub Actions in addition to Circle CI and look how it works.

### PR DESCRIPTION
**Pull request by bershanskiy**
Fixes mozilla/addons#1469. Run a subset of CI jobs on GitHub Actions in addition to Circle CI. Please see linked issue for motivation.

Thanks for opening a Pull Request (PR), here's a few guidelines as to what we need in your PR before we review it.

Please delete anything that isn't relevant to your patch.

- [x] This PR relates to an existing open issue and there are no existing PRs open for the same issue.
- [x] Add `Fixes #ISSUENUM` at the top of your PR.
- [x] Add a description of the changes introduced in this PR.
- [x] The change has been successfully run locally.
- [x] Add tests to cover the changes added in this PR.

Once you have met the above requirements please replace this section with a `Fixes #ISSUENUM` linking to the issue fixed by this PR along with an explanation of the changes. Thanks for your contribution!

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+name: Run tests on PR
+
+on:
```
Comment (bershanskiy) [inline .github/workflows/test.yml:3]: If you like, I can change to just `workflow_dispatch` to let users run GitHub Actions manually but conserve electricity and GitHub Actions credits if they do not run tests manually.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
+
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 19.x
```
Comment (bershanskiy) [inline .github/workflows/test.yml:26]: Using a single version of Node to conserve resources.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      - synchronize
+      - reopened
+      - ready_for_review
+  push:
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Run tests
+    runs-on: ubuntu-latest
```
Comment (bershanskiy) [inline .github/workflows/test.yml:18]: Using standard GitHub Actions image instead of standard Circle CI image.

**Comment by willdurand** [comment]
I am going to close this PR for now since we are not ready to accept it.

**Pull request closed without merging**